compiler/rustc_session/src/search_paths.rs

```diff
@@ -1,12 +1,27 @@
 use crate::filesearch::make_target_lib_path;
 use crate::{config, early_error};
+use std::cmp::Ordering;
 use std::path::{Path, PathBuf};
 
 #[derive(Clone, Debug)]
 pub struct SearchPath {
     pub kind: PathKind,
     pub dir: PathBuf,
-    pub files: Vec<SearchPathFile>,
+    // njn: comments, esp. that these are sorted. Also that the division is
+    // suboptimal, because Target isn't available yet
+    // njn: ideal would be rlibs, rmetas, dylibs, staticlibs, others
+
+    // `.rlib` files in this directory.
+    pub rlibs: Vec<SearchPathFile>,
+
+    // `.rmeta` files in this directory.
+    pub rmetas: Vec<SearchPathFile>,
```

Comment:
The ordering in which the files are generally considered is rmeta -> rlib -> dylib (-> garbage), from the cheapest to more expensive, I suggest using it everywhere.